angularjs/angular.d.ts

```diff
@@ -756,7 +756,15 @@ declare namespace angular {
          * @return Html-linkified and sanitized text.
          * see https://docs.angularjs.org/api/ngSanitize/filter/linky
          */
-        (text: string, target: string, attributes:Object|Function): string;
+        <T>(text: string, target: string, attributes?:IFilterLinkyAttributesMapObject|IFilterLinkyAttributesFunc<T>): string;
```

Comment:
Interfaces aren't needed for something as simple as `{[attribute: string]: string}`. Also the function shouldn't be generic and it returns an object not a string. 

> `function`: Takes the url as a parameter and returns a map of attributes

So the type of this parameter is:
```ts
attributes?: { [attribute: string]: string } | ((url: string) => { [attribute: string]: string })
```